pkg/controller/controller.go

```diff
@@ -98,7 +101,7 @@ type Controller struct {
 
 // NewMeshController builds the informers and other required components of the mesh controller, and returns an
 // initialized mesh controller object.
-func NewMeshController(clients k8s.Client, cfg Config, logger logrus.FieldLogger) (*Controller, error) {
+func NewMeshController(clients k8s.Client, cfg Config, store SharedStore, logger logrus.FieldLogger) (*Controller, error) {
```

Comment:
Should this return the stopChan as well as the Controller? so that we can use the stop chan to stop the controller, instead of having to use the context to do so?

Reply:
I found an alternative solution, removing the need of a `Stop` method and the need of a taking a context in the `Run` method.